hnn_core/network.py

```diff
@@ -423,15 +418,14 @@ def _create_all_spike_sources(self):
 
                 # new ExtFeed, where now both feed type and target cell type
                 # specified because these feeds have cell-specific parameters
-                # XXX unique_feeds is a dict of dict
-                self.unique_feeds[src_type].append(
-                    ExtFeed(feed_type=src_type,
-                            target_cell_type=target_cell_type,
-                            params=self.p_unique[src_type],
-                            gid=gid))
-                pc.cell(gid,
-                        self.unique_feeds[src_type][-1].connect_to_target(
-                            self.params['threshold']))
+                unique_feed = ExtFeed(feed_type=src_type,
+                                      target_cell_type=target_cell_type,
+                                      params=self.p_unique[src_type],
+                                      gid=gid)
+                self._artificial_cells.append(
+                    ArtificialCell(unique_feed.event_times,
+                                   self.params['threshold']))
+                pc.cell(gid, self._artificial_cells[-1].nrn_netcon)
```

Comment:
can we rename `_artificial_cells` to be `_feed_cells` to make it clear that these cells relate to the external feed?

Now we have this incongruity that the class `ExtFeed` does not really create a "feed" but just gives us event_times. I guess it will be fixed once we separate the class properly

Reply:
Sure, we can do that. So maybe the feed module will eventually contain a single function, callable by the `Network` class, that creates instances of `ExtFeed` and `ArtificialCell` to be passed into `pc.cell()`?

Reply:
right, but for command-line interface, I think we need to give user the ability to create feed objects outside of the `Network` class and then the `Network` class will simply create the `_feed_cells` from these objects.